pkg/kademlia/lookup.go

```diff
@@ -4,89 +4,135 @@
 package kademlia
 
 import (
-	"bytes"
 	"context"
 	"log"
-	"time"
+	"math/big"
+	"sync"
 
 	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/node"
 	"storj.io/storj/pkg/pb"
 )
 
-type sequentialLookup struct {
-	contacted       map[string]bool
-	queue           *XorQueue
-	slowestResponse time.Duration
-	client          node.Client
-	target          dht.NodeID
-	limit           int
-	bootstrap       bool
+type concurrentLookup struct {
```

Comment:
rename to `discovery`

Reply:
Is that a term that's used in any kad spec language? I think there's still a better term if not; personally, I prefer `lookup` over `discovery.

Reply:
My main problem with lookup is that we have the `lookup` in a few places, as previously discussed in Slack. Of course, I'm open to a better name than `discovery`. Alternatively we can try find better names for the other things.

Reply:
There's also term location used here https://en.wikipedia.org/wiki/Kademlia#Locating_nodes.

Reply:
It seems some implementations have used the term `crawl`